File: PaymentWithoutChange/awnser.py
import sys


def can_pay_exact(a: int, b: int, n: int, s: int) -> bool:
    max_n_coins = min(a, s // n)
    remaining = s - max_n_coins * n
    return remaining <= b


def main() -> None:
    data = sys.stdin.read().strip().split()
    if not data:
        return

    q = int(data[0])
    idx = 1

    out_lines = []
    for _ in range(q):
        a = int(data[idx])
        b = int(data[idx + 1])
        n = int(data[idx + 2])
        s = int(data[idx + 3])
        idx += 4

        if can_pay_exact(a, b, n, s):
            out_lines.append("YES")
        else:
            out_lines.append("NO")

    sys.stdout.write("\n".join(out_lines))


if __name__ == "__main__":
    main()

File: PaymentWithoutChange/awnser.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func canPayExact(a, b, n, s int64) bool {
	maxNCoins := a
	if s/n < maxNCoins {
		maxNCoins = s / n
	}

	remaining := s - maxNCoins*n
	return remaining <= b
}

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	var q int
	if _, err := fmt.Fscan(in, &q); err != nil {
		return
	}

	for i := 0; i < q; i++ {
		var a, b, n, s int64
		fmt.Fscan(in, &a, &b, &n, &s)

		if canPayExact(a, b, n, s) {
			fmt.Fprintln(out, "YES")
		} else {
			fmt.Fprintln(out, "NO")
		}
	}
}
